docs/operator-manual/applicationset/Appset-Any-Namespace.md

```diff
@@ -0,0 +1,177 @@
+# ApplicationSet in any namespace
+
+**Current feature state**: Beta
+
+!!! warning
+    Please read this documentation carefully before you enable this feature. Misconfiguration could lead to potential security issues.
+
+## Introduction
+
+As of version 2.7, Argo CD supports managing `ApplicationSet` resources in namespaces other than the control plane's namespace (which is usually `argocd`), but this feature has to be explicitly enabled and configured appropriately.
+
+Argo CD administrators can define a certain set of namespaces where `ApplicationSet` resources may be created, updated and reconciled in. 
+
+As Applications generated by an ApplicationSet are generated in the same namespace as the ApplicationSet itself, this works in combination with [App in any namespace](../app-any-namespace.md).
+    
+## Prerequisites
+
+### App in any namespace configured
+
+This feature needs [App in any namespace](../app-any-namespace.md) feature activated. The list of namespace must be the same.
+
+### Cluster-scoped Argo CD installation
+
+This feature can only be enabled and used when your Argo CD ApplicationSet controller is installed as a cluster-wide instance, so it has permissions to list and manipulate resources on a cluster scope. It will *not* work with an Argo CD installed in namespace-scoped mode.
+
+## Implementation details
+
+### Overview
+
+In order for an ApplicationSet to be managed and reconciled outside the Argo CD's control plane namespace, two prerequisites must match:
+
+1. The `ApplicationSet`'s namespace must be explicitly enabled using the `--applicationset-namespaces` parameter for the `argocd-applicationset-controller`. This parameter controls the list of namespaces that Argo CD will be allowed to source `ApplicationSet` resources from globally.
+2. The number of namespaces covered by the `--applicationset-namespaces` must be entirely covered by the [App in any namespace](../app-any-namespace.md), otherwise the generated Applications generated outside the allowed Application namespaces won't be reconciled.
```

Comment:
I think we should encourage people to use `application.namespaces` in argocd-cmd-params-cm instead of directly setting `--applicationset-namespaces`. Setting applicationset namespaces to a subset of application namespaces can be a secondary recommendation.

Reply:
How can I update the codegen part so that environment variable `ARGOCD_APPLICATIONSET_NAMESPACES` is declared in Deployment `argocd-applicationset-controller` (ie: `manifests/install.yaml`) and mapped to `application.namespaces` in `argocd-cmd-params-cm` ?

Reply:
The base manifest has to be updated manually (you can copy, paste, and modify an existing environment variable in the base statefulset) and then run `make manifests-local`.

Reply:
Done